### PR TITLE
Automate jenkins.io changelog

### PR DESCRIPTION
**Pull request by timja**
See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

Automates generating changelog on every merge to master

It will either keep pushing to a branch (with open PR) or if people have been merging the changes it will create a new PR, (note: manual copyedits will be overridden), the proposed changelog entries themselves should be updated, or copyediting should be done after the release is complete as the script will move on to the next release once the tag exists

This has been end 2 end tested on my fork,

See:
https://github.com/timja/jenkins/runs/2875832053?check_suite_focus=true
https://github.com/timja/jenkins.io/pull/3

Requires:
https://github.com/jenkinsci/core-changelog-generator/pull/14
and
https://issues.jenkins.io/browse/INFRA-2900

### Proposed changelog entries

* Entry 1: Not needed
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: .github/workflows/changelog.yml
Comment (timja) [inline .github/workflows/changelog.yml:42]: this section will need updating once the github app is created

File: .github/workflows/changelog.yml
Comment (timja) [inline .github/workflows/changelog.yml:44]: branch and repo need updating after merge of https://github.com/jenkinsci/core-changelog-generator/pull/14

File: .github/workflows/changelog.yml
Comment (timja) [inline .github/workflows/changelog.yml:30]:
```suggestion
          repository: jenkins-infra/jenkins.io
```

File: .github/workflows/changelog.yml
Comment (timja) [inline .github/workflows/changelog.yml:28]:
```suggestion
          app_id: ${{ secrets.JENKINS_CHANGELOG_UPDATER_APP_ID }}
```

File: .github/workflows/changelog.yml
Comment (timja) [inline .github/workflows/changelog.yml:29]:
```suggestion
          private_key: ${{ secrets.JENKINS_CHANGELOG_UPDATER_PRIVATE_KEY }}
```

**Review by basil — APPROVED** [review]

**Comment by daniel-beck** [comment]
If we're losing the ability to copyedit and curate the changelog, what is the point of still having it, instead of using GitHub's releases API?

**Comment by timja** [comment]
> If we're losing the ability to copyedit and curate the changelog, what is the point of still having it, instead of using GitHub's releases API?

You can still copy edit it either after the release is done and do some final clean-ups, or copy edit the actual changelog entry in the pull request and it will be preserved.

Swapping entirely to GitHub releases at least for weekly is another option, the main bit we would lose I guess is the changelog weather feedback which can be quite good although is often spammed.

**Comment by daniel-beck** [comment]
> You can still copy edit it either after the release is done and do some final clean-ups

Let's be real here, nobody is going to bother doing that. Right now, we have no changelog, which means it's sort of important someone creates it, and then it is no effort to do it well. The threshold for taking something that exists and change a phrasing for clarity is much higher.

**Comment by MarkEWaite** [comment]
> > You can still copy edit it either after the release is done and do some final clean-ups

> Let's be real here, nobody is going to bother doing that. Right now, we have no changelog, which means it's sort of important someone creates it, and then it is no effort to do it well. The threshold for taking something that exists and change a phrasing for clarity is much higher.

We discussed this in Docs office hours and @dheerajodha, @stackscribe, and @kwhetstone felt that we would willingly do the copy editing in the pull request from this automation.

File: .github/workflows/changelog.yml
Comment (oleg-nenashev) [inline .github/workflows/changelog.yml:28]: I would rather prefer to keep it for now until the flow is fully settled. I use this asset in one of my experiments (changleog => opengraph generator)

**Review by oleg-nenashev — APPROVED** [review]
I am +1 for it. It is a non-destructive change for jenkins.io . Would be nice to get responses from @MarkEWaite @daniel-beck though.

Bonus points for keeping the release asset for now until the new flow is fully adopted

**Comment by oleg-nenashev** [comment]
https://github.com/jenkinsci/core-changelog-generator/pull/14 is merged now @timja

**Comment by timja** [comment]
Let's try it out

**Review by oleg-nenashev — APPROVED** [review]
+1 for shipping it. 

We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.

We can also merge them now if others agree

**Pull request merged**
